### DISCUSSION TITLE
Smooth Pan and zoom

### DISCUSSION THREAD

**Discussion by deicidemilan** [chartjs/chartjs-plugin-zoom#605] [Q&A]
Hi,

Maybe this is not right place to set question, but i try to get pan and zoom on chart.js 3 as on v2, and can't get it.
Even on official page [Official example](https://www.chartjs.org/chartjs-plugin-zoom/latest/guide/usage.html) chart is broken if I try to zoom in, in local pan is broken too.
Did I miss something? I have this example on v2 and works perfect. [v2 example](https://jsfiddle.net/1fjnq96r)
I upload and v3 example. After some time if I pan drag right, Chart does not behave expected [v3 example](https://jsfiddle.net/hewfotLu/)
If this is not possible on v3 I need to downgrade to v2 :(

Best regards,
Milan

**Comment by LeeLenaleee** [comment]
Pan is not broken in the official sample you linked. Instead it is dissabled. The zoom and pan are by default disabled and have to be enabled for all the modes you want it to use.

For a pan and zoom example you can see this example: https://www.chartjs.org/chartjs-plugin-zoom/latest/samples/basic.html

**Edit:**
You are comparing 2 verry different charts for v2 against v3. Your v2 example uses a timescale which has a whole lot of precision because of how it is implemented and can thus scroll seemingly smooth. While you in your v3 example use a category scale which cant devide in smaller sub section. Also if you zoom first so not the whole chart is shown in your v3 example you will see the pan is working fine

**Comment by deicidemilan** [comment]
@LeeLenaleee I did't understand and implement plugin correctly. Thanks for explain.
Sorry for misunderstand lib and plugin how works together.

**Comment by LeeLenaleee** [comment]
np can happen
